weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/ManageColumnsButton.tsx

```diff
@@ -0,0 +1,192 @@
+/**
+ * Button/popup for grid column visibility state.
+ */
+
+import {Popover} from '@mui/material';
+import {GridColDef, GridColumnVisibilityModel} from '@mui/x-data-grid-pro';
+import {Switch} from '@wandb/weave/components';
+import classNames from 'classnames';
+import React, {useCallback, useRef, useState} from 'react';
+
+import {Button} from '../../../../../Button';
+import {DraggableGrow} from '../../../../../DraggablePopups';
+import {TextField} from '../../../../../Form/TextField';
+import {Tailwind} from '../../../../../Tailwind';
+import {ColumnInfo} from '../../types';
+import {TraceCallSchema} from '../wfReactInterface/traceServerClient';
+
+type ManageColumnsButtonProps = {
+  columnInfo: ColumnInfo;
+  columnVisibilityModel: GridColumnVisibilityModel;
+  setColumnVisibilityModel: (model: GridColumnVisibilityModel) => void;
+};
+
+export const ManageColumnsButton = ({
+  columnInfo,
+  columnVisibilityModel,
+  setColumnVisibilityModel,
+}: ManageColumnsButtonProps) => {
+  const [search, setSearch] = useState('');
+  const ref = useRef<HTMLDivElement>(null);
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const onClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(anchorEl ? null : ref.current);
+    setSearch('');
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popper' : undefined;
+
+  const toggleColumnShow = useCallback(
+    (key: string) => {
+      const on = columnVisibilityModel[key] ?? true;
+      const newModel = {
+        ...columnVisibilityModel,
+        [key]: !on,
+      };
+      setColumnVisibilityModel(newModel);
+    },
+    [columnVisibilityModel, setColumnVisibilityModel]
+  );
+  const onHideAll = () => {
+    const newModel = columnInfo.cols.reduce((acc, col) => {
+      if (col.hideable ?? true) {
+        acc[col.field] = false;
+      }
+      return acc;
+    }, {} as GridColumnVisibilityModel);
+    setColumnVisibilityModel(newModel);
+  };
+  const onShowAll = () => {
```

Comment:
is `showAll` the default? When I click `showAll` I get the constructed link, which is quite large, we could just use the vanilla link with no filter options right? Otherwise storing the info in the link is soo nice.

Reply:
for example, to me, these two links show an identical table: 

```http://localhost:9000/griffin_wb/prod-evals-july/weave/traces```

```http://localhost:9000/griffin_wb/prod-evals-july/weave/traces?cols=%7B%22feedback%22%3Atrue%2C%22derived.status_code%22%3Atrue%2C%22inputs.self%22%3Atrue%2C%22inputs.model%22%3Atrue%2C%22output.score_answer_format.correct.true_count%22%3Atrue%2C%22output.score_answer_format.correct.true_fraction%22%3Atrue%2C%22output.score_answer_expected.correct.true_count%22%3Atrue%2C%22output.score_answer_expected.correct.true_fraction%22%3Atrue%2C%22output.model_latency.mean%22%3Atrue%2C%22wb_user_id%22%3Atrue%2C%22started_at%22%3Atrue%2C%22derived.tokens%22%3Atrue%2C%22derived.cost%22%3Atrue%2C%22derived.latency%22%3Atrue%7D```

Reply:
Good question! Right now, all columns are shown by default. However, not that long ago we had to disable a nice feature we used to have (was going to be complicated with server side pagination), where we would automatically hide columns that were mostly blank. I'm not sure if we will bring that back or not. I also think it is quite possible that we will introduce other columns that are hidden by default, for example, we might want to add separate columns for Input Tokens and Output Tokens - we probably don't need to show both of those by default in addition to Total tokens, but they could become available via this dialog.

If we do get to a situation where not all columns are shown by default, we'd probably want to add another action like "Reset". So we'll probably need to revisit no matter what.

While keeping the URL short would be nice, it's already unreadable as soon as you open a peek drawer, and it's about to get even worse once we start keeping filter state etc in there. (Which will be a massive UX improvement other than URL length.)

So I'm inclined to keep it as is, but if you think it would be better for Show all to reset to default I'm OK with that too.